compiler/fir/tree/src/org/jetbrains/kotlin/fir/FirEffectiveVisibility.kt

```diff
@@ -45,7 +45,8 @@ interface FirEffectiveVisibility {
         }
 
         override fun lowerBound(other: FirEffectiveVisibility): FirEffectiveVisibility {
-            throw AssertionError("Should not be called")
+            // This default object acts as TOP in the lattice, hence by-passing `other` as a result of `meet`.
+            return other
```

Comment:
Maybe I'm wrong? Is it rather `BOTTOM`? In either case, for the least lower bound computation, I guess it's safe to return `other` as-is here.

Reply:
Obsolete comment.